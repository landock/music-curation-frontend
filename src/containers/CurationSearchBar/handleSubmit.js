import configureActionAndDispatchIt from './configureActionAndDispatchIt';
import { map, withLatestFrom } from 'rxjs/operators';

export default function handleSubmit(selectorStream, stateStream, propsStream) {
  console.log(selectorStream.pipe, '\n', stateStream, '\n', propsStream);
  let stream = selectorStream.pipe(
    withLatestFrom(stateStream),
    map(([e, state]) => state),
    withLatestFrom(propsStream),
    map(([state, props]) => {
      const { searchTerm } = state;
      return configureActionAndDispatchIt(props, searchTerm);
    })
  );
  return stream;
}
