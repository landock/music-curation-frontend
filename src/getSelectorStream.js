export default function getSelectorStream(
  selectorFunc,
  selectorArg,
  eventName
) {
  if (!selectorFunc) return;
  return selectorFunc(selectorArg).addListener(eventName);
}
