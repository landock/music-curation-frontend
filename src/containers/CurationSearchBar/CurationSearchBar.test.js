import * as CurationSearchBarModule from '.';
import { of } from 'rxjs/observable/of';

import getSelectorStream from '../../getSelectorStream';
import handleSubmit from './handleSubmit';

jest.mock('../../getSelectorStream');
jest.mock('./handleSubmit');

describe('CurationSearchBar', () => {
  let { view, dispatch } = CurationSearchBarModule;
  describe('view', () => {
    it('match snapshot', () => {
      const searchTerm = 'beagles al dente';
      const mockProps = {};
      const mockState = { searchTerm };
      const result = new view(mockProps, mockState);
      expect(result).toMatchSnapshot();
    });
  });
  describe('dispatch', () => {
    it('should call getSelectorStream and handleSubmit and return the result as an array', () => {
      const searchTerm = 'beagles al dente';

      getSelectorStream.mockReturnValue(of('whatever'));
      const state = of({ id: 1 });
      const mockSources = {
        select: of('www'),
        state,
        props: of({}),
      };
      CurationSearchBarModule.handleSubmit = jest.fn();

      const result = new dispatch(mockSources);

      expect(getSelectorStream).toHaveBeenCalled();
      expect(handleSubmit).toHaveBeenCalled();
    });
  });
});
