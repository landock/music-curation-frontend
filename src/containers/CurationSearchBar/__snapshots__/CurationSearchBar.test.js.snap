// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurationSearchBar view match snapshot 1`] = `
<Form
  as="form"
  id="search-form"
>
  <FormInput
    as={[Function]}
    control={[Function]}
    icon="search"
    id="search-term"
    placeholder="Search Curations"
    value="beagles al dente"
  />
</Form>
`;
